nsxt/resource_nsxt_edge_transport_node.go

```diff
@@ -671,43 +677,55 @@ func getTransportNodeFromSchema(d *schema.ResourceData, m interface{}) (*mpmodel
 	description := d.Get("description").(string)
 	displayName := d.Get("display_name").(string)
 	tags := getMPTagsFromSchema(d)
+	nodeID := d.Get("node_id").(string)
 	failureDomain := d.Get("failure_domain").(string)
 	hostSwitchSpec, err := getHostSwitchSpecFromSchema(d, m, nodeTypeEdge)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create Transport Node: %v", err)
 	}
 
-	converter := bindings.NewTypeConverter()
-	var dataValue data.DataValue
-	var errs []error
+	var nodeDeploymentInfo *data.StructValue
+	if nodeID == "" {
+		/*
+			node_id attribute conflicts with node_deployment_info. As node_deployment_info is a complex object which has
```

Comment:
Could you please clarify - even when specifying `node_id`, user still needs to populate `node_deployment_info`? What properties are needed?
I don't think `node_deployment_info` is `Computed`

Reply:
The resource has no `node_deployment_info` attribute as it's not implementing the entire API, as host nodes are created with policy API. Relevant `node_deployment_info` attributes are placed under `deployment_config`, and directly under the resource root.
I've set these attributes as conflicting with `node_id`.